code/client/src/context/authContext.js

```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+const AuthContext = React.createContext(false);
```

Comment:
Instead of exporting just `AuthContext`, you can also define here something like

```jsx
const AuthContext = createContext([{}, () => {}])

const AuthProvider({children}) => {
   const [user, setUser] = useState(null)
   const [loggedIn, setLoggedIn] = useState(false)
   const [loading, setLoading] = useState(false)
   const [dirty, setDirty] = useState(false)

   useEffect(() => {
      // APIs and logic to check if the user is logged in or not.
      // In case, user is logged in you can also save user info
   }, [dirty])

   if (loading) return <Spinner />
   else return (
      <AuthContext.Provider value={[{...user, loggedIn}, setLoggedIn]}>
          {children}
      <AuthContext.Provider />
   )
}

export {AuthContext, AuthProvider}
```

In this way, where you need to check if the user is logged in, you just have to import the context and with, e.g.

```jsx
const [user, updateRequest] = useContext(AuthContext)
```

or if you don't need to request updates also

```jsx
const [user] = useContext(AuthContext)
```